syntax = "proto3";
option go_package = "/protocol/app";



message Command {
    uint64 id = 1;
    repeated string args = 2;
    repeated string env = 3;
    bytes stdin = 4;
    int64 timeoutMillis = 5;
}

message CommandResult {
    uint64 commandId = 1;
    int32 exitCode = 2;
    bytes stdout = 3;
    bytes stderr = 4;
}

message CommandListMsg {
    repeated Command commands = 1;
}

message CommandResultListMsg {
    repeated CommandResult commandResults = 1;
}

message GetCommandListMsg {}

message AgentMsg {
    oneof MsgType {
        CommandResultListMsg CommandResultListMsg = 1;
        GetCommandListMsg GetCommandListMsg = 2;
    }
}

message SuccessMsg {}

message LPMsg {
    oneof MsgType {
        SuccessMsg SuccessMsg = 1;
        CommandListMsg CommandListMsg = 2;
    }
}